### ISSUE TITLE
Ctrl+Shift+LMB Won't Cut single strip

### ISSUE THREAD

**Issue by doakey3** [GDquest/Blender-power-sequencer#118]
If I do Ctrl+LMB, I can trim part of a single channel. If I do Ctrl+Shift+LMB, the function affects all channels. I think these two functions should be similar in behavior.

[GIF Example](https://i.imgur.com/K1JFvz7.gif)

**Comment by NathanLovato** [comment]
If you Ctrl LMB on a strip only - if you click on an empty channel it'll trim all channels. These ops are super important for video editing. Ctrl here means cut/trim and Shift means force all (or time cursor mode). Just like backspace removes one gap and shift backspace should remove all of them, C concatenate one strip, Shift C all strips in channel.
If you have ideas for better keymaps though there's a task open for it #23 so don't hesitate to make suggestions

**Comment by doakey3** [comment]
Why not make it so that if you Ctrl+Shift+LMB on empty channel space you affect all strips (as you do with Ctrl+LMB) and only affect 1 strip when you have the mouse on that strip. This seems intuitive because it's the same behavior as the Ctrl+LMB

**Comment by NathanLovato** [comment]
As I said, Shift is meant to mean "all" (channels, strips...) across the keymaps so it's coherent in that sense.
More importantly the cursor mode is essential for editing as the smart mode means you'll work slower: you have to move the cursor out of the strips every time you want to make a cut.

Now I have worked too much with the add-on to really see some of the UX issues so if it feels awkard as it is, how about:

- Make Shift Ctrl LMB work as you suggest, using the auto detection mode and remove gaps (gotta add support for this on a single strip)
- Add Shift LMB as a replacement for the cursor mode override. The shortcut is still not in use.

What do you think? We could also use some code to better detect if the user clicks on a strip or not. Right now it relies on Blender's view to region function but even though the strip take only 2/3rds of a channel it only treats clicks in channel. Wish I had the time to get to all that

**Comment by NathanLovato** [comment]
I worked on it and I don't find it an improvement. I've tried adding Alt to the mix as it's blender's key for "ripple edit", doing it without it but in either case you lose efficiency or meaningful shortcuts (Ctrl for cut, Shift for all - like Shift C to concatenate all in channel, etc.).

We can rework the UX, give it more thought, but changing the shortcuts didn't improve the interactions to me.
